plugins/modules/ipatrust.py

```diff
@@ -0,0 +1,282 @@
+            msg="Trust add has failed for '%s'" % (realm))
+    else:
+        return None
+
+
+def gen_args(trust_type, realm_admin, realm_passwd, realm_server, trust_secret, base_id, range_size, range_type, bidirectional, external):
+    _args = {}
+    if trust_type is not None:
+        _args["trust_type"] = trust_type
+    if realm_admin is not None:
+        _args["realm_admin"] = realm_admin
+    if realm_passwd is not None:
+        _args["realm_passwd"] = realm_passwd
+    if realm_server is not None:
+        _args["realm_server"] = realm_server
+    if trust_secret is not None:
+        _args["trust_secret"] = trust_secret
+    if base_id is not None:
+        _args["base_id"] = base_id
+    if range_size is not None:
+        _args["range_size"] = range_size
+    if bidirectional is not None:
+        _args["bidirectional"] = bidirectional
+    if external is not None:
+        _args["external"] = external
+
+    return _args
+
+
+def main():
+    ansible_module = AnsibleModule(
+        argument_spec=dict(
+            # general
+            ipaadmin_principal=dict(type="str", default="admin"),
+            ipaadmin_password=dict(type="str", required=False, no_log=True),
+            realm=dict(type="str", default=None, required=True),
+            # state
+            state=dict(type="str", default="present",
+                       choices=["present", "absent"]),
+            # present
+            trust_type=dict(type="str", default="ad", required=False),
+            admin=dict(type="str", default=None, required=False),
+            passwordd=dict(type="str", default=None,
+                           required=False, no_log=True),
+            server=dict(type="str", default=None, required=False),
+            trust_secret=dict(type="str", default=None,
+                              required=False, no_log=True),
+            base_id=dict(type="int", default=None, required=False),
+            range_size=dict(type="int", default=200000, required=False),
+            range_type=dict(type="str", default="ipa-ad-trust", required=False, choices=[
+                            "ipa-ad-trust-posix", "ipa-ad-trust"]),
+            two_way=dict(type="bool", default=False, required=False),
+            external=dict(type="bool", default=False, required=False),
+        ),
+        supports_check_mode=True,
+    )
+
+    ansible_module._ansible_debug = True
+
+    # general
+    ipaadmin_principal = module_params_get(
+        ansible_module, "ipaadmin_principal")
+    ipaadmin_password = module_params_get(ansible_module, "ipaadmin_password")
+    realm = module_params_get(ansible_module, "realm")
+
+    # state
+    state = module_params_get(ansible_module, "state")
+
+    # trust
+    trust_type = module_params_get(ansible_module, "trust_type")
+    realm_admin = module_params_get(ansible_module, "admin")
+    realm_passwd = module_params_get(ansible_module, "password")
+    realm_server = module_params_get(ansible_module, "server")
```

Comment:
For consistency, which would ease maintainance, with other modules, you should use the same variable name as the parameter whenever it is possible.

Reply:
done